pkg/dsl/events.go

```diff
@@ -138,26 +146,30 @@ func UponNodeSigsVerified[C any](
 			return nil
 		}
 
-		var nodeIds []t.NodeID
-		for _, id := range ev.NodeIds {
-			nodeIds = append(nodeIds, t.NodeID(id))
+		nodeIds := make([]t.NodeID, len(ev.NodeIds))
+		for i := range ev.NodeIds {
+			nodeIds[i] = t.NodeID(ev.NodeIds[i])
 		}
 
-		var errs []error
-		for _, err := range ev.Errors {
-			errs = append(errs, errors.New(err))
+		errs := make([]error, len(ev.Valid))
```

Comment:
Is this copying intended for better isolation?
Otherwise, the `ev.Errors` slice could be used directly.
If you do want a copy and errors are actually values and not pointers (not 100%), you can also simply use `copy` to create a new error slice instead of looping through the errors.

Reply:
It's for the sake of type conversion. `ev.Errors` is a slice of strings and I need a slice of errors (with nil indicating that no error). Note that the handler doesn't accept the `ev.Valid` slice. In some sense, the information from `ev.Valid []bool` and `ev.Errors []string` is "merged" into `errs []error`, to make it more idiomatic go.

Reply:
Right, makes sense. In fact, I've been hesitating removing the slice of `bool`s anyway. As you probably noticed, the `allOk` field is also just sugar. But that seems to be rather convenient, so I'm keeping it there too.